### ISSUE TITLE
reactive config loading buggy

### ISSUE THREAD

**Issue by nasser** [arcadia-unity/Arcadia#237]
Did not pick up my config changes after editing `Assets/configuration.edn`. Hard to say why or provide an illustrative example.

**Comment by timsgardner** [comment]
Need more than this

**Comment by timsgardner** [comment]
I've found something that might be an issue.
After compiling core, it looks like we're redefining clojure.core.protocols/CollReduce. The filewatcher uses arcadia.internal.map-utils/valsr, which reifies that protocol. The redefinition breaks the reification.

Reloading the VM after compiling core should do the trick. Removing the dependency on valsr ought to do it too.

**Comment by timsgardner** [comment]
Though if we're also compiling map-utils it shouldn't be a problem... nevertheless, after compiling I end up with:
```clojure
(instance? clojure.core.protocols.CollReduce
  (arcadia.internal.map-utils/valsr
    (zipmap (range 10) (range))))
;; => false
```
then after reevaling map-utils:
```clojure
(instance? clojure.core.protocols.CollReduce
  (arcadia.internal.map-utils/valsr
    (zipmap (range 10) (range))))
;; => true
```
If this is a problem here, it will be a problem elsewhere.

I notice we're compiling internal namespaces one at a time, this seems dangerous; let's use `arcadia.compiler/aot-namespaces` instead

**Comment by timsgardner** [comment]
I'm assuming this one is closed as of a946a4fd517be085c92e40d874ffa4dbd8837fbb, which changes the aot policy such that namespaces aren't double-aot'd.
